src/lib/utils.ts

```diff
@@ -0,0 +1,124 @@
+import type { JSONSchema, SchemaObject } from 'oas/dist/rmoas.types';
+
+import { get as lodashGet } from 'lodash'; // eslint-disable-line no-restricted-imports
+
+/**
+ * Determine if a schema `type` is, or contains, a specific discriminator.
+ *
+ */
+export function hasSchemaType(
+  schema: SchemaObject,
+  discriminator: 'array' | 'object' | 'string' | 'number' | 'boolean' | 'integer' | 'null'
+) {
+  if (Array.isArray(schema.type)) {
+    return schema.type.includes(discriminator);
+  }
+
+  return schema.type === discriminator;
+}
+
+/**
+ * With a supplied JSON Schema object, spider through it for any schemas that may contain specific
+ * kind of `format` that also happen to be within the current `requestBody` payload that we're
+ * creating a HAR representation for.
+ *
+ */
+export function getTypedFormatsInSchema(
+  format: 'json' | 'binary',
+  schema: any,
+  opts: {
+    parentIsArray?: boolean;
+    parentKey?: string;
+    payload: any;
+  }
+): boolean | string | (string | boolean)[] {
+  try {
+    if (schema?.format === format) {
+      if (opts.parentIsArray) {
+        const parentData = lodashGet(opts.payload, opts.parentKey);
+        if (parentData !== undefined && Array.isArray(parentData)) {
+          return Object.keys(parentData)
+            .map(pdk => {
+              const currentKey = [opts.parentKey, pdk /* , key */].join('.');
```

Comment:
I don't really understand the `key` comment, is that just an artifact of development?

Reply:
Artifact yeah, will remove.